src/signTx_ui.c

```diff
@@ -409,94 +602,441 @@ void signTx_handleCertificate_ui_runStep()
+			#endif // HAVE_BAGL
+			break;
+
+		case CERTIFICATE_STAKE_DEREGISTRATION:
+		case CERTIFICATE_STAKE_DEREGISTRATION_CONWAY:
+			#ifdef HAVE_BAGL
+			snprintf(description, SIZEOF(description), "deregistration?");
+			#elif defined(HAVE_NBGL)
+			display_confirmation("Confirm\nderegistration", "", "DEREGISTRATION\nACCEPTED", "Deregistration\nrejected", this_fn, respond_with_user_reject);
+			#endif // HAVE_BAGL
+			break;
+
+		case CERTIFICATE_STAKE_DELEGATION:
 			#ifdef HAVE_BAGL
-			ui_displayPathScreen(
-			        "Stake key",
-			        &BODY_CTX->stageData.certificate.stakeCredential.keyPath,
+			snprintf(description, SIZEOF(description), "delegation?");
+			#elif defined(HAVE_NBGL)
+			display_confirmation("Confirm\ndelegation", "", "DELEGATION\nACCEPTED", "Delegation\nrejected", this_fn, respond_with_user_reject);
+			#endif // HAVE_BAGL
+			break;
+
+		default:
+			ASSERT(false);
+		}
+		// make sure all the information is displayed to the user
+		ASSERT(strlen(description) + 1 < SIZEOF(description));
+
+		#ifdef HAVE_BAGL
+		ui_displayPrompt(
+		        "Confirm",
+		        description,
+		        this_fn,
+		        respond_with_user_reject
+		);
+		#elif defined(HAVE_NBGL)
+		#endif // HAVE_BAGL
+	}
+	UI_STEP(HANDLE_CERTIFICATE_STAKING_STEP_RESPOND) {
+		respondSuccessEmptyMsg();
+
+		tx_advanceCertificatesStateIfAppropriate();
+	}
+	UI_STEP_END(HANDLE_CERTIFICATE_STAKING_STEP_INVALID);
+}
+
+void signTx_handleCertificateVoteDeleg_ui_runStep()
+{
+	TRACE("UI step %d", ctx->ui_step);
+	ui_callback_fn_t* this_fn = signTx_handleCertificateVoteDeleg_ui_runStep;
+	sign_tx_certificate_data_t* cert = &BODY_CTX->stageData.certificate;
+
+	UI_STEP_BEGIN(ctx->ui_step, this_fn);
+
+	UI_STEP(HANDLE_CERTIFICATE_VOTE_DELEGATION_STEP_DISPLAY_OPERATION) {
+		#ifdef HAVE_BAGL
+		ui_displayPaginatedText(
+		        "Delegate",
+		        "vote",
+		        this_fn
+		);
+		#elif defined(HAVE_NBGL)
+		set_light_confirmation(true);
+		display_prompt("Delegate\nvote", "", this_fn, respond_with_user_reject);
+		#endif // HAVE_BAGL
+	}
+	UI_STEP(HANDLE_CERTIFICATE_VOTE_DELEGATION_STEP_DISPLAY_STAKE_CRED) {
+		_displayCredential(
+		        this_fn,
+		        &cert->stakeCredential,
+		        "Stake key",
+		        "Stake key hash",
+		        "stake_vkh",
+		        "Stake script hash",
+		        "script"
+		);
+	}
+	UI_STEP(HANDLE_CERTIFICATE_VOTE_DELEGATION_STEP_DISPLAY_DREP) {
+		switch (cert->drep.type) {
+		case EXT_DREP_KEY_PATH:
+			_displayKeyPath(this_fn, &cert->drep.keyPath, "DRep key");
+			break;
+		case EXT_DREP_KEY_HASH:
+			_displayKeyHash(this_fn, cert->drep.keyHash, "DRep key hash", "drep");
+			break;
+		case EXT_DREP_SCRIPT_HASH:
+			_displayScriptHash(this_fn, cert->drep.scriptHash, "DRep script hash", "drep");
+			break;
+		case DREP_ALWAYS_ABSTAIN:
+			#ifdef HAVE_BAGL
+			ui_displayPaginatedText(
+			        "Always",
+			        "abstain",
```

Comment:
I'd keep this one because it describes what will happen (as a consequence of this delegation).
For UI details, there is no one that would give us some definitive guidance, everyone just suggests this and that and I can't evaluate the proposals due to lack of information. So at this point, we want to put something out (on experimental provider) and let SW wallets & M. Lang try it and then gather feedback.

Reply:
Ok, let's do it consistently on Trezor then.

cc @jaskp (only changing the string in UI should be sufficient)